mobile-browser-based-version/server/src/logic/federated/handlers.js

```diff
@@ -439,47 +450,47 @@ export function postWeights(request, response) {
 export async function aggregationStatus(request, response) {
   const type = REQUEST_TYPES.AGGREGATION_STATUS;
 
-  const code = _checkRequest(request);
-  if (code !== 200) {
-    return _failRequest(response, type, code);
-  }
-
   const task = request.params.task;
   const round = request.params.round;
   const id = request.params.id;
 
-  /**
-   * The task was not trained at all.
-   */
-  if (
-    !tasksStatus.get(task).isRoundPending &&
-    tasksStatus.get(task).round === 0
-  ) {
-    return _failRequest(response, type, 403);
+  if (!clients.has(id)) {
+    return _failRequest(response, type, 401);
+  }
+  if (!tasksStatus.has(task)) {
+    return _failRequest(response, type, 404);
   }
-  /**
-   * No weight was posted for this task's round.
-   */
   if (!(weightsMap.has(task) && weightsMap.get(task).has(round))) {
```

Comment:
should return aggregated=false instead of 404, otherwise late clients might get disconnected (in addition to receiving weird 404 errors although everything is normal)